lib/api/src/grpc/conversions.rs

```diff
@@ -939,7 +939,11 @@ fn conditions_helper_to_grpc(conditions: Option<Vec<segment::types::Condition>>)
             if conditions.is_empty() {
                 vec![]
             } else {
-                conditions.into_iter().map(|c| c.into()).collect()
+                conditions
+                    .into_iter()
+                    .filter(|c| !matches!(c, segment::types::Condition::Resharding(_))) // TODO(resharding)!?
+                    .map(|c| c.into())
+                    .collect()
```

Comment:
A bit hacky, but this was *much* easier, than refactoring all read operations, so that we only propagate `Condition::Resharding` to local read operations, but not to remote read operations.

Reply:
Would a `condition.is_local_only()` make sense here?